### PR TITLE
7 post article comment

### PR DESCRIPTION
**Pull request by milbot1992**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -80,4 +80,79 @@ describe('All wrong paths', () => {
...
+            expect(res.body.comment).toMatchObject({
+                                                        comment_id: 19,
+                                                        body: 'great article',
+                                                        article_id: 1,
+                                                        author: 'butter_bridge',
+                                                        votes: 0,
+                                                        created_at: expect.any(String)
+                                                    })
+        })
+    });
```
Comment (B7Bprog) [inline __tests__/app.test.js:170]: Please add a test (201) to check if extra properties in the posted object are ignored. For example when sending this:
`{
                            username: 'butter_bridge',
                            body: 'great article',
                            extraProp: 'extraValue'
                            }`

File: __tests__/app.test.js [hunk 2]
```diff
@@ -80,4 +80,79 @@ describe('All wrong paths', () => {
...
+    test('should return 400 Bad Request if given an article_id that does not exist',()=>{
```
Comment (B7Bprog) [inline __tests__/app.test.js]: This supposed to be a 404, Not found, where the article with that id doesn't exist but can be added anytime. Please refactor this test.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -80,4 +80,79 @@ describe('All wrong paths', () => {
...
+            }
+        return request(app)
+        .post('/api/articles/notAnId/comments')
+        .send(newComment)
+        .expect(400)
+        .then(({body})=>{
+            expect(body.message).toBe('Invalid ID')
+        })
+    })
+    test('should return a 406 Not acceptable if the object passed is incorrectly formatted',()=>{
```
Comment (B7Bprog) [inline __tests__/app.test.js]: We haven't got a list of acceptable values to choose from here in our server, and it's still just a bad request, which should be 400 here.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -80,4 +80,79 @@ describe('All wrong paths', () => {
...
+            }
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(406)
+        .then ((res)=>{
+            expect(res.body.message).toBe('Not acceptable, request missing required columns')
+        })
+    })
+    test('should return a 400 Bad request if the object passed has bad values - username must appear in users table to be accepted',()=>{
```
Comment (B7Bprog) [inline __tests__/app.test.js]: This should test for 404 not found, where the username has a valid type, but doesn't exist in the database. Please change this test as well.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -80,4 +80,79 @@ describe('All wrong paths', () => {
...
+            username: 'milbot1992',
+            body: 'fab article'
+            }
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(400)
+        .then ((res)=>{
+            expect(res.body.message).toBe('Bad request')
+        })
```
Comment (B7Bprog) [inline __tests__/app.test.js:254]: Please add another test for a bad request where the post object is missing required properties:

`{
                            username: 'butter_bridge',
                  
                            }`

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Millie, please see my comments and refactor your code. Once it's all done and tests are passing, please send this PR with the updates back to us for a review.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -129,30 +150,42 @@ describe('POST /api/articles/:article_id/comments', () => {
             expect(body.message).toBe('Invalid ID')
         })
     })
-    test('should return a 406 Not acceptable if the object passed is incorrectly formatted',()=>{
+    test('should return a 400 Bad Request if the object passed is incorrectly formatted',()=>{
```
Comment (Augs0) [inline __tests__/app.test.js]: Not sure if this test overlaps with the next one with a missing property. Also, the comment does not look badly formatted

**Review by Augs0 — APPROVED** [review]
Nice work implementing those changes. I think you may have a duplicate test in there at the moment, but I'll leave you to have a peek at this. You're ok to merge.

**Pull request merged**